### DISCUSSION TITLE
Rancher Desktop Design - About

### DISCUSSION THREAD

**Discussion by rak-phillip** [rancher-sandbox/rancher-desktop#689] [Archive]
### About

* Rename "General" to "About"
* De-emphasize this menu option by moving to bottom of navigation

![2 - About](https://user-images.githubusercontent.com/835961/134972119-4efdfcee-350e-4773-b754-322c3ecc791d.png)

> Using the space to share this initial design proposal from the UI group. I've add some quick notes for each section of Rancher Desktop that might change. If you have any questions about anything you see below, please ask and I'll provide some more detail. 
>
> Original Discussion #683
>
> Feedback is highly valued and welcome.

**Comment by jandubois** [comment]
> ### About
> * Rename "General" to "About"
> * De-emphasize this menu option by moving to bottom of navigation

I would prefer to rename it to "Welcome" and leave it at the top.

Whenever RD terminates, it should store both the current version, and the currently active page in settings.

When you start RD, it compares the now current version against the stored version. If they are different, the starting page should be "Welcome". Otherwise it should be the same page that was shown when the app was closed.

This gives us a nice and friendly welcoming page, that the user is revisiting after each app update, so we can show the newly available features, or whatever else we may need to point out.

But during regular work, the "current" tab is maintained during app restarts, so the "Welcome" page doesn't get in the way.

Comment copied from https://github.com/rancher-sandbox/rancher-desktop/discussions/683#discussioncomment-1393094

**Reply by rak-phillip** [reply]
I don't think it's the right choice to keep About/Welcome at the top of the list. A common practice is to keep the most frequently used/important menu items placed at the top to promote recall and ease of access. Most design systems will promote a similar pattern. If we consider [Material Design navigation drawers](https://material.io/components/navigation-drawer#usage) as one data point, for example:

> Organized
>
> Navigation drawers order destinations according to user importance, with frequent destinations first and related ones grouped together.

Other design systems will make similar call-outs, and I don't see Welcome or About being something that will be of great importance in day-to-day usage to warrant the top stop in our navigation. 

With that said, ends of lists are important and easily accessible as well. The end of the list could serve as an excellent place for less commonly used functions like preferences, about, or info; I think it would be a good idea to stick to this convention.

**Comment by jandubois** [comment]
RD is neither a mobile nor a web app, so I'm not sure those "ease of access" rules apply here. An obvious counter-example is that "About App" is the top menu choice in each app menu:

<img width="210" alt="Screen Shot 2021-09-28 at 1 20 54 PM" src="https://user-images.githubusercontent.com/78947/135159926-8f8a41d7-cf29-4eaa-a947-9c483e67364c.png">

 Even the [material.io/design](https://material.io/design) app shows "Introduction" at the coveted top left spot, instead of e.g. "Components", which the frequent user will refer to much more often.

This should be the first screen that a new user sees (and the first screen you see after each upgrade), so putting it at the end just feels wrong. As a new user, I expect to go through the pages sequentially to set up the app the way I want it to work.

Anyways, I do feel more strongly about showing the "About" or "Welcome" page again after each upgrade, and keeping the current page active across app restarts, then about the location.

However, I would be strongly opposed to dynamically re-arranging the order of the pages based on usage patterns, so that the most used page floats to the top (as implied by "according to user importance"). Navigations must stay in a predictable place and not move around over time.


<img width="290" alt="Screen Shot 2021-09-28 at 1 01 09 PM" src="https://user-images.githubusercontent.com/78947/135158941-a735b3f2-496a-439d-8f6a-def144e9a0de.png">

**Reply by rak-phillip** [reply]
Even in the Safari example, About is separate from the primary mode of navigation. Would a better solution be to access About/Welcome solely via the app menu (this is typically accessed via Help on Windows & Linux)? 

> Anyways, I do feel more strongly about showing the "About" or "Welcome" page again after each upgrade, and keeping the current page active across app restarts, then about the location.

No argument here, this is a well-defined pattern that plenty of desktop apps exhibit after an update.  

> However, I would be strongly opposed to dynamically re-arranging the order of the pages based on usage patterns, so that the most used page floats to the top (as implied by "according to user importance"). Navigations must stay in a predictable place and not move around over time.

Agreed, but being in alpha state, I'm hoping that we would be a little more amenable to change as we level set on some patterns. As it stands, having the first item in the left navigation be something that a user will skip over a majority of the time to get to actions that they might want to take, like scanning an image or modifying resources, just seems a little off to me. 

Perhaps, Material Design was a bad example to raise. Once again, it's not the only design system out there that makes such a recommendation. If we look at other desktop apps, they certainly do not provide much importance to the notion of welcome/about in the primary navigation. For example, Slack:

![Screen Shot 2021-09-28 at 2 00 41 PM](https://user-images.githubusercontent.com/835961/135164989-4df86a3e-fe79-4ca2-b2dc-2d27979d6213.png)

VSCode:

![Screen Shot 2021-09-28 at 2 01 51 PM](https://user-images.githubusercontent.com/835961/135165132-4d78254d-2269-44f1-b8d1-3b863ceef6a2.png)

Or even macOS apps, like Finder:

![Screen Shot 2021-09-28 at 2 03 07 PM](https://user-images.githubusercontent.com/835961/135165311-77bc0b59-62db-4bee-b17b-804727724b3c.png)

Notes:

![Screen Shot 2021-09-28 at 2 04 39 PM](https://user-images.githubusercontent.com/835961/135165532-5757f760-1a60-4396-b913-3ab7566cf792.png)

**Reply by rak-phillip** [reply]
I'd like to emphasize that this is great feedback and I thank you for the discussion. 

If the team thinks that Welcome/About or whatever we call it stays at the top, I am willing to concede and move along with the choice. I just think there is a better use for such prime real estate in the navigation.

**Reply by jandubois** [reply]
> Agreed, but being in alpha state, I'm hoping that we would be a little more amenable to change as we level set on some patterns.

Yes, I think we have the freedom to move things around for a while.

Maybe I misread the "according to user importance" part, but I assumed you would measure the time spend on (or just how often a page would be selected), and the UI would dynamically rearrange the navigation based on the usage pattern. So it would look different for each user, and also could change for a user when their habits change. And this strikes me as a very bad idea, but now I think they probably didn't even mean that. 😄

It probably means "according to what the developer thinks is of most importance to the user", with all users expected to use the app in a very similar manner.

**Comment by mook-as** [comment]
This was mentioned in other channels, but that large blue hero image looks very space-wasting. It's possible that this is an artifact of the sample images being of _very_ large windows, though; I would not expect Rancher Desktop to be often maximized, as people are not likely to be using it exclusively / immersed for long periods (they're much more likely to be living in their editors, browsers, or command line).

I assume some of our users would be using 13 inch or so laptops, on a train (or wherever external monitors are impractical), hacking along as they commute.  At 16:9 that's ~11" wide, or ~ 70em (*very* roughly at 12pt — this assume 1 css inch = 1 physical inch).

**Reply by rak-phillip** [reply]
Agreed. I think the idea was to add a little flair, but it's probably not needed.
